logstash-core/lib/logstash/pipeline.rb

```diff
@@ -447,12 +447,14 @@ def worker_loop(batch_size, batch_delay)
       shutdown_requested |= signal.shutdown? # latch on shutdown signal
 
       batch = @filter_queue_client.read_batch # metrics are started in read_batch
-      @events_consumed.increment(batch.size)
-      filter_batch(batch)
-      flush_filters_to_batch(batch, :final => false) if signal.flush?
-      output_batch(batch)
-      break if @force_shutdown.true? # Do not ack the current batch
-      @filter_queue_client.close_batch(batch)
+     if (batch.size > 0)
+        @events_consumed.increment(batch.size)
+        filter_batch(batch)
+        flush_filters_to_batch(batch, :final => false) if signal.flush?
+        output_batch(batch)
+        break if @force_shutdown.true? # Do not ack the current batch
```

Comment:
I don't _think_ we can currently get into condition of empty batch and force_shutdown.  But that may not always be true, so I updated the logic in https://github.com/elastic/logstash/pull/7559/commits/32633a7cad9af6567a1d47a033f5e5c18e5538f4 (please double check the logic).  I will squash and re-base before a merge.